source/easeOfAccess.py

```diff
@@ -1,36 +1,57 @@
 # A part of NonVisual Desktop Access (NVDA)
-# Copyright (C) 2014-2021 NV Access Limited
+# Copyright (C) 2014-2022 NV Access Limited
 # This file is covered by the GNU General Public License.
 # See the file COPYING for more details.
 
 """Utilities for working with the Windows Ease of Access Center.
 """
 
+from enum import Enum, IntEnum
+from typing import List
+from logHandler import log
 import winreg
-import ctypes
 import winUser
 import winVersion
 
+
 # Windows >= 8
 canConfigTerminateOnDesktopSwitch: bool = winVersion.getWinVer() >= winVersion.WIN8
+_APP_KEY_NAME = "nvda_nvda_v1"
+
+
+class RegistryKey(str, Enum):
+	ROOT = r"Software\Microsoft\Windows NT\CurrentVersion\Accessibility"
+	TEMP = r"Software\Microsoft\Windows NT\CurrentVersion\AccessibilityTemp"
+	APP = r"%s\ATs\%s" % (ROOT, _APP_KEY_NAME)
+
+
+class AutoStartContext(IntEnum):
+	"""Registry HKEY used for tracking when NVDA starts automatically"""
+	ON_LOGON_SCREEN = winreg.HKEY_LOCAL_MACHINE
+	AFTER_LOGON = winreg.HKEY_CURRENT_USER
 
-ROOT_KEY = r"Software\Microsoft\Windows NT\CurrentVersion\Accessibility"
-APP_KEY_NAME = "nvda_nvda_v1"
-APP_KEY_PATH = r"%s\ATs\%s" % (ROOT_KEY, APP_KEY_NAME)

```

Comment:
Maybe the constants should be kept for backwards compat. They could be wrapped with a deprecation version check, give addon devs more time to remove usages.

Reply:
I'm happy to do that, and set them to be deprecated in 2023.1. However, our API is yet to be finalised anyway, as we have several API breaking changes posed still (#13007, #12710).

Reply:
Where possible we should try to give add-on authors advanced warning for changes to the API, if doing so is low cost for maintenance. In this case I think it is. There will certainly be other things that break the API in this release, by deprecating, we are giving add-on authors a full year to update / migrate away from the deprecations.

Reply:
I've fixed this in 91143ce. I agree that this is low cost for maintenance and can be delayed without problem.